innovations/views.py

```diff
@@ -1,3 +1,61 @@
-from django.shortcuts import render
+from django.contrib.auth.decorators import login_required
+from django.db.models import Model
+from django.http import Http404
+
+from django.shortcuts import render, redirect
+
+from innovations.models import Innovation
+from signup.groups import administrators, committee_members, in_groups
+
+
+@login_required
+def my_innovations(request):
+    innovations = get_filtered_innovations(issuer=request.user)
+    return render(request, "innovations/innovations_list.html", {"innovations": innovations})
+
+
+@login_required
+def with_status(request, status):
+    if is_forbidden(status, request.user):
+        raise Http404("Page not found!")
+    innovations = get_filtered_innovations(status=status)
+    return render(request, "innovations/innovations_list.html", {"innovations": innovations})
+
+
+@login_required
+def single(request, id):
+    try:
+        innovation = Innovation.objects.get(id=id)
+    except Model.DoesNotExist:
+        raise Http404("Page not found!")
+    if is_forbidden(innovation.status, request.user):
+        raise Http404("Page not found!")
+    return render(request, "innovations/innovations_list.html", {"innovations": [innovation]})
+
+
+@login_required
+def set_status(request, id, status):
+    if not has_confidential_access(request.user):
+        raise render(request, "permission_denied.html")
+    innovation = Innovation.objects.get(id=id)
+    innovation.status = status
+    innovation.save()
```

Comment:
Lepiej: `Innovation.objects.filter(id=id).update(status=status)` Będzie jedna linijka, a django to zrobi w jednym zapytaniu nie pobierając nic z bazy i nie tworząc niepotrzebnego obiektu.